### DISCUSSION TITLE
Generate posterior expectations / predictions for new data

### DISCUSSION THREAD

**Discussion by stefgehrig** [liesel-devs/liesel#183] [Q&A] [answered]
Is there a simple way to create posterior predictions / posterior expectations of predictions from a model object and the fitted model *after* sampling from the posterior (i.e., after running something like `engine.sample_all_epochs()`)? The intention would be to supply a new data frame, similar as is possible in R with many model types via `predict(model, newdata = …)`.

In `{brms}`, for example, there is such functionality via `brms::posterior_predict()` and `brms::posterior_epred()`, respectively.

I'm not a firm Python user, but this would probably be best done within Python?

**Comment by jobrachem** [comment]
I'm afraid that, currently, Liesel does not offer much convenience functionality for predictions. You get the posterior samples, but have to implement custom functionality for predictions. We have an issue for that here: https://github.com/liesel-devs/liesel/issues/54 But unfortunately I don't expect that we will have the time to prioritise prediction convenience in the near future.

**Reply by stefgehrig** [reply]
Makes total sense, it is not a must-have. Thank you!

**Comment by jobrachem** [comment, marked as answer]
@stefgehrig in the new release `v0.4.0`, Liesel now has prediction convenience :) Here's a link to the method documentation: [`lsl.Model.predict`](https://docs.liesel-project.org/en/latest/generated/liesel.model.Model.predict.html#liesel.model.Model.predict) and [`lsl.Var.predict`](https://docs.liesel-project.org/en/latest/generated/liesel.model.Var.predict.html#liesel.model.Var.predict)

**Reply by stefgehrig** [reply]
Amazing work, thank you for pursuing this (:
